blakserv/webhook.c

```diff
@@ -0,0 +1,222 @@
+// Meridian 59, Copyright 1994-2012 Andrew Kirmse and Chris Kirmse.
+// All rights reserved.
+//
+// This software is distributed under a license that is described in
+// the LICENSE file that accompanies it.
+//
+// Meridian is a registered trademark.
+/*
+* webhook.c
+*
+
+  Cross-platform webhook message delivery via named pipes (Windows) or FIFOs (Linux/macOS).
+  Keeps pipes open for performance instead of opening/closing on every message.
+
+*/
+
+#include "blakserv.h"
+#include "webhook.h"
+#include <time.h>
+
+#ifdef BLAK_PLATFORM_WINDOWS
+#include <windows.h>
+#else
+#include <fcntl.h>
+#include <unistd.h>
+#endif
+
+// Configuration constants
+static const int MAX_WEBHOOK_PIPES_WINDOWS = 10;
+static const int MAX_WEBHOOK_PIPES_UNIX = 10;
+
+// Global state for persistent pipe connections
+static bool webhook_initialized = false;
+static char pipe_prefix[64] = "";
+static int last_pipe_index = 0;
+
+#ifdef BLAK_PLATFORM_WINDOWS
+static HANDLE pipe_handles[10];
```

Comment:
Use MAX constants rather than hard-coding 10 again.

Also, you should try hard to reduce the number of #ifdefs.  For example, osd_linux.h already defines HANDLE to int, so you can probably just use the Windows version here on both platforms.